lib/packwerk/const_node_inspector.rb

```diff
@@ -29,11 +26,15 @@ def constant_name_from_node(node, ancestors:)
 
     private
 
+    # Only process the root `const` node for namespaced constant references. For example, in the
+    # reference `Spam::Eggs::Thing`, we only process the const node associated with `Spam`.
+    def root_constant?(parent)
+      !parent || !Node.constant?(parent)
```

Comment:
For me this would be 1% easier to understand as `!(parent && Node.constant?(parent))` (per [de Morgan](https://en.wikipedia.org/wiki/De_Morgan%27s_laws)) but there’s not much in it.